server/src/main/java/io/seata/server/storage/file/lock/FileLocker.java

```diff
@@ -77,27 +89,39 @@ public boolean acquireLock(List<RowLock> rowLocks) {
             int bucketId = pk.hashCode() % BUCKET_PER_TABLE;
             BucketLockMap bucketLockMap = CollectionUtils.computeIfAbsent(tableLockMap, bucketId,
                 key -> new BucketLockMap());
-            Long previousLockTransactionId = bucketLockMap.get().putIfAbsent(pk, transactionId);
-            if (previousLockTransactionId == null) {
-                //No existing lock, and now locked by myself
+            BranchSession previousLockBranchSession = bucketLockMap.get().putIfAbsent(pk, branchSession);
+            if (previousLockBranchSession == null) {
+                // No existing lock, and now locked by myself
                 Set<String> keysInHolder = CollectionUtils.computeIfAbsent(bucketHolder, bucketLockMap,
                     key -> ConcurrentHashMap.newKeySet());
                 keysInHolder.add(pk);
-            } else if (previousLockTransactionId == transactionId) {
+            } else if (previousLockBranchSession.getTransactionId() == transactionId) {
                 // Locked by me before
                 continue;
             } else {
-                LOGGER.info("Global lock on [" + tableName + ":" + pk + "] is holding by " + previousLockTransactionId);
+                LOGGER.info("Global lock on [" + tableName + ":" + pk + "] is holding by " + previousLockBranchSession.getBranchId());
                 try {
                     // Release all acquired locks.
                     branchSession.unlock();
                 } catch (TransactionException e) {
                     throw new FrameworkException(e);
                 }
-                return false;
+                if (!autoCommit && previousLockBranchSession.getLockStatus() == LockStatus.Rollbacking) {
+                    failFast = true;
+                    break;
+                }
+                if (canLock) {
```

Comment:
how to process？
autoCommit=false&&previousLockBranchSession.getLockStatus()=Locked